src/legacy/core_plugins/metrics/public/components/panel_config/timeseries.js

```diff
@@ -42,21 +42,28 @@ import {
   EuiHorizontalRule,
 } from '@elastic/eui';
 import { injectI18n, FormattedMessage } from '@kbn/i18n/react';
-
+import { Storage } from 'ui/storage';
+import { data } from 'plugins/data';
+import chrome from 'ui/chrome';
+const uiSettingsQueryLanguage = chrome.getUiSettingsClient().get('search:queryLanguage');
+const { QueryBarInput } = data.query.ui;
+const localStorage = new Storage(window.localStorage);
 class TimeseriesPanelConfigUi extends Component {
-
   constructor(props) {
     super(props);
     this.state = { selectedTab: 'data' };
   }
 
+  handleQueryChange = filter => {
```

Comment:
nip: not sure that I'm right but for me it looks like extra method.  What about
`onChange={filter => this.props.onChange({ filter })}`

Reply:
done